src/visitors.rs

```diff
@@ -1882,26 +2164,36 @@ impl<'a, 'b: 'a, 'tcx: 'b, E> MirVisitor<'a, 'b, 'tcx, E> {
         if let mir::Place::Projection(boxed_place_projection) = place {
             let base_ty = self.get_rustc_place_type(&boxed_place_projection.base);
             match boxed_place_projection.elem {
-                mir::ProjectionElem::Deref => {
-                    debug!("base_ty: {:?}", base_ty);
-                    match base_ty {
-                        TyKind::Adt(..) => base_ty,
-                        TyKind::RawPtr(ty_and_mut) => &ty_and_mut.ty.sty,
-                        TyKind::Ref(_, ty, _) => &ty.sty,
-                        _ => unreachable!(),
+                mir::ProjectionElem::Deref => match base_ty {
+                    TyKind::Adt(..) => base_ty,
+                    TyKind::RawPtr(ty_and_mut) => &ty_and_mut.ty.sty,
+                    TyKind::Ref(_, ty, _) => &ty.sty,
+                    _ => {
+                        println!("span: {:?}", self.current_span);
```

Comment:
Did you mean to leave these printlns in here?

Reply:
Yes, I am by no means convinced that these cases are actually unreachable, so if they actually occur, I'd like to see a bit more information than just there is a bug in Mirai. I could put them all in a string message and pass it to unreachable, but that seems a little less readable to me.

Reply:
Why not use the logger then?